### PR TITLE
change test, add retention_size, minifix config

### LINKED ISSUE #197: Bug:  fix plugin/output/s3/s3_test.go
some tests fall oft, check why and make minifix

### PR DESCRIPTION
**Pull request by DSmolonogov**
Fixes #197

### CODE REVIEW

File: plugin/output/file/file.go [hunk 1]
```diff
@@ -188,6 +197,9 @@ func (p *Plugin) fileSealUpTicker(ctx context.Context) {
 	for {
 		timer := time.NewTimer(time.Until(p.nextSealUpTime))
 		select {
+		case <-p.retentionChan:
+			timer.Stop()
```
Comment (ansakharov) [inline plugin/output/file/file.go:201]: Why stopping timer? Sealing file rotates with new one

File: pipeline/streamer.go [hunk 2]
```diff
@@ -49,8 +49,6 @@ func (s *streamer) start() {
 }
 
 func (s *streamer) stop() {
-	s.shouldStop.Store(true)
```
Comment (ansakharov) [inline pipeline/streamer.go:52]: How it stops now?

File: pipeline/streamer.go [hunk 3]
```diff
@@ -148,6 +145,7 @@ func (s *streamer) resetBlocked(stream *stream) {
 
 func (s *streamer) heartbeat() {
 	streams := make([]*stream, 0)
+	s.shouldStop.Store(false)
```
Comment (ansakharov) [inline pipeline/streamer.go:148]: Why stopping in heartbeat?

**Review by ansakharov — CHANGES_REQUESTED** [review]

**Review by vadimalekseev — APPROVED** [review]

File: plugin/output/file/file.go [hunk 4]
```diff
@@ -254,12 +272,13 @@ func (p *Plugin) sealUp() {
 	oldFile := p.file
 	p.mu.Lock()
 	p.createNew()
-	p.nextSealUpTime = time.Now().Add(p.config.RetentionInterval_)
+	p.fileIsFully = false
```
Comment (vano144) [inline plugin/output/file/file.go:275]: what do you think about move this (initialization of p.fileIsFully and p.fileSize) in createNew ?

File: plugin/output/file/file.go [hunk 5]
```diff
@@ -244,6 +261,7 @@ func (p *Plugin) sealUp() {
 	if err != nil {
 		p.logger.Panicf("could not get info about file: %s, error: %s", p.file.Name(), err.Error())
 	}
+	p.nextSealUpTime = time.Now().Add(p.config.RetentionInterval_)
```
Comment (vano144) [inline plugin/output/file/file.go:264]: Please add comment why it sould be before check on size

File: plugin/output/file/file.go [hunk 6]
```diff
@@ -191,6 +200,8 @@ func (p *Plugin) fileSealUpTicker(ctx context.Context) {
 	for {
 		timer := time.NewTimer(time.Until(p.nextSealUpTime))
 		select {
+		case <-p.retentionChan:
```
Comment (vano144) [inline plugin/output/file/file.go:203]: what about case when both options are on?

**Comment by ansakharov** [comment]
lgtm

**Review by ansakharov — APPROVED** [review]

**Comment by vadimalekseev** [comment]
Closed due to exceeding dependency caching quota for open PR

**Pull request closed without merging**